pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -175,6 +174,7 @@ extraAfter=(${hardeningCFlagsAfter[@]+"${hardeningCFlagsAfter[@]}"} $NIX_CFLAGS_
 extraBefore=(${hardeningCFlagsBefore[@]+"${hardeningCFlagsBefore[@]}"} $NIX_CFLAGS_COMPILE_BEFORE_@suffixSalt@)
 
 if [ "$dontLink" != 1 ]; then
+    linkType=$(checkLinkType $NIX_LDFLAGS_BEFORE_@suffixSalt@ "${params[@]}" ${NIX_CFLAGS_LINK_@suffixSalt@:-} $NIX_LDFLAGS_@suffixSalt@)
```

Comment:
This doesn't look quite right to me — if `linkType` contains other flags, checks like this one below won't make sense:

```bash
    if [[ "$linkType" == dynamic && -n "$NIX_DYNAMIC_LINKER_@suffixSalt@" ]]; then
```

Reply:
$linkType does not contain flags, it is either `static` or `dynamic` according to the output of [`checkLinkType()`](https://github.com/NixOS/nixpkgs/blob/697312fb824243bd7bf82d2a3836a11292614109/pkgs/build-support/wrapper-common/utils.bash#L138-L149)

Reply:
Yep, my mistake.